delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/ElasticSearchViews.scala

```diff
@@ -1,21 +1,254 @@
+    for {
+      p   <- projects.fetchActiveProject(project)
+      iri <- expandIri(id, p)
+      res <- eval(TagElasticSearchView(iri, project, tagRev, tag, rev, subject), p)
+    } yield res
+  }.named("tagElasticSearchView", moduleType)
+
+  def deprecate(
+      id: IdSegment,
+      project: ProjectRef,
+      rev: Long
+  )(implicit subject: Subject): IO[ElasticSearchViewRejection, ElasticSearchViewResource] = {
+    for {
+      p   <- projects.fetchActiveProject(project)
+      iri <- expandIri(id, p)
+      res <- eval(DeprecateElasticSearchView(iri, project, rev, subject), p)
+    } yield res
+  }.named("deprecateElasticSearchView", moduleType)
+
+  def fetch(id: IdSegment, project: ProjectRef): IO[ElasticSearchViewRejection, ElasticSearchViewResource] =
+    fetch(id, project, None).map(_._1).named("fetchElasticSearchView", moduleType)
+
+  def fetchAt(
+      id: IdSegment,
+      project: ProjectRef,
+      rev: Long
+  ): IO[ElasticSearchViewRejection, ElasticSearchViewResource] =
+    fetch(id, project, Some(rev)).map(_._1).named("fetchElasticSearchViewAt", moduleType)
+
+  private def fetch(
+      id: IdSegment,
+      project: ProjectRef,
+      rev: Option[Long]
+  ): IO[ElasticSearchViewRejection, (ElasticSearchViewResource, Iri)] =
+    for {
+      p     <- projects.fetchProject(project)
+      iri   <- expandIri(id, p)
+      state <- rev.fold(currentState(project, iri))(stateAt(project, iri, _))
+      res   <- IO.fromOption(state.toResource(p.apiMappings, p.base), ViewNotFound(iri, project))
+    } yield (res, iri)
+
+  def fetchBy(
+      id: IdSegment,
+      project: ProjectRef,
+      tag: Label
+  ): IO[ElasticSearchViewRejection, ElasticSearchViewResource] =
+    fetch(id, project, None)
+      .flatMap { case (resource, iri) =>
+        resource.value.tags.get(tag) match {
+          case Some(rev) => fetchAt(id, project, rev).leftMap(_ => TagNotFound(tag, iri, project))
+          case None      => IO.raiseError(TagNotFound(tag, iri, project))
+        }
+      }
+      .named("fetchElasticSearchViewByTag", moduleType)
+
+  def list(
+      pagination: FromPagination,
+      params: ElasticSearchViewSearchParams
+  ): UIO[UnscoredSearchResults[ElasticSearchViewResource]] =
+    cache.values
+      .map { resources =>
+        val results = resources.filter(params.matches).toVector.sortBy(_.createdAt)
+        UnscoredSearchResults(
+          results.size.toLong,
+          results.map(UnscoredResultEntry(_)).slice(pagination.from, pagination.from + pagination.size)
+        )
+      }
+      .named("listElasticSearchViews", moduleType)
+
+  def events(offset: Offset): fs2.Stream[Task, Envelope[ElasticSearchViewEvent]] =
+    eventLog.eventsByTag(moduleType, offset)
+
+  private def currentState(project: ProjectRef, iri: Iri): IO[ElasticSearchViewRejection, ElasticSearchViewState] =
+    aggregate.state(identifier(project, iri))
+
+  private def stateAt(project: ProjectRef, iri: Iri, rev: Long) = {
```

Comment:
There is an operation provided by EventLogUtils that makes this a little shorter

Reply:
it's not available... it's defined in the service

Reply:
Maybe move it to the sdk?

Reply:
the EventLog is not available in the sdk... so the utils can't be moved.